vendor/views/vendor.py

```diff
@@ -200,7 +192,7 @@ def post(self, request, *args, **kwargs):
 
         shipping_address = form.save(commit=False)
 
-        invoice = get_purchase_invoice(request.user)
+        invoice = get_purchase_invoice(request.user, request.site)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
invoice = get_purchase_invoice(request.user, get_site_from_request(request))